app/services/push_notifications/send.rb

```diff
@@ -23,7 +25,7 @@ def call
       # has a constraint that will execute the job only once (30 seconds from now). So if we need to send 1 PN
       # every 5 seconds we only execute this once every 30s (no duplicate/unnecessary processing).
       # If no PNs are scheduled to be sent for a 6h span then 0 jobs are executed.
-      PushNotifications::DeliverWorker.perform_in(30.seconds)
+      PushNotifications::DeliverWorker.perform_in(30.seconds) if relation.any?
```

Comment:
We do one additional query BUT we also avoid to queue a worker in case there are no devices with those user ids.

I wonder if the previous scenario was intentional though.

cc @fdoxyz

Reply:
Yeah, we don't have data yet. That said, we can always remove it if we have millions of devices ;-)